From: Debian PHP Maintainers <pkg-php-maint@lists.alioth.debian.org>
Date: Fri, 16 Oct 2015 18:49:09 +0200
Subject: Disable apxs2 module activation on sapi install

---
 sapi/apache2handler/config.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sapi/apache2handler/config.m4 b/sapi/apache2handler/config.m4
index f170ea9..3b280dd 100644
--- a/sapi/apache2handler/config.m4
+++ b/sapi/apache2handler/config.m4
@@ -69,7 +69,7 @@ if test "$PHP_APXS2" != "no"; then
                 \$(mkinstalldirs) '$APXS_SYSCONFDIR' && \
                  $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
                        -S SYSCONFDIR='$APXS_SYSCONFDIR' \
-                       -i -a -n php5"
+                       -i -n php5"
   fi
 
   case $host_alias in
