From: =?utf-8?b?T25kxZllaiBTdXLDvQ==?= <ondrej@sury.org>
Date: Wed, 13 May 2020 08:35:54 +0200
Subject: Backport the aarch64 ZEND_SIGNED_MULTIPLY_LONG assembly fix

---
 Zend/zend_multiply.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Zend/zend_multiply.h b/Zend/zend_multiply.h
index a4d48b1..850c108 100644
--- a/Zend/zend_multiply.h
+++ b/Zend/zend_multiply.h
@@ -62,8 +62,8 @@
 	__asm__("mul %0, %2, %3\n"										\
 		"smulh %1, %2, %3\n"										\
 		"sub %1, %1, %0, asr #63\n"									\
-			: "=X"(__tmpvar), "=X"(usedval)							\
-			: "X"(a), "X"(b));										\
+			: "=&r"(__tmpvar), "=&r"(usedval)						\
+			: "r"(a), "r"(b));										\
 	if (usedval) (dval) = (double) (a) * (double) (b);				\
 	else (lval) = __tmpvar;											\
 } while (0)
