From: Remi Collet <remi@remirepo.net>
Date: Mon, 28 Jun 2021 14:31:02 +0200
Subject: adapt for 5.6 without ZEND_ULONG_MAX

---
 ext/pdo_firebird/firebird_statement.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/pdo_firebird/firebird_statement.c b/ext/pdo_firebird/firebird_statement.c
index dc64c19..ea3c704 100644
--- a/ext/pdo_firebird/firebird_statement.c
+++ b/ext/pdo_firebird/firebird_statement.c
@@ -275,7 +275,7 @@ static int firebird_fetch_blob(pdo_stmt_t *stmt, int colno, char **ptr, /* {{{ *
 		ISC_STATUS stat;
 
 		/* prevent overflow */
-		if (*len == ZEND_ULONG_MAX) {
+		if (*len == (LONG_MAX * 2UL +1UL)) {
 			result = 0;
 			goto fetch_blob_end;
 		}
