From: =?utf-8?b?T25kxZllaiBTdXLDvQ==?= <ondrej@sury.org>
Date: Tue, 11 Mar 2025 16:26:11 +0100
Subject: Force using POSIX threads on Linux

---
 TSRM/threads.m4 | 46 ++--------------------------------------------
 1 file changed, 2 insertions(+), 44 deletions(-)

diff --git a/TSRM/threads.m4 b/TSRM/threads.m4
index 38494ce..6ce395e 100644
--- a/TSRM/threads.m4
+++ b/TSRM/threads.m4
@@ -109,50 +109,8 @@ dnl  -qthreaded        AIX cc V5
 dnl  -threads          gcc (HP-UX)
 dnl
 AC_DEFUN([PTHREADS_CHECK],[
-
-if test "$beos_threads" = "1"; then
-  pthreads_working="yes"
-  ac_cv_pthreads_cflags=""
-else
-  save_CFLAGS=$CFLAGS
-  save_LIBS=$LIBS
-  PTHREADS_ASSIGN_VARS
-  PTHREADS_CHECK_COMPILE
-  LIBS=$save_LIBS
-  CFLAGS=$save_CFLAGS
-
-  AC_CACHE_CHECK(for pthreads_cflags,ac_cv_pthreads_cflags,[
-  ac_cv_pthreads_cflags=
-  if test "$pthreads_working" != "yes"; then
-    for flag in -kthread -pthread -pthreads -mthreads -Kthread -threads -mt -qthreaded; do 
-      ac_save=$CFLAGS
-      CFLAGS="$CFLAGS $flag"
-      PTHREADS_CHECK_COMPILE
-      CFLAGS=$ac_save
-      if test "$pthreads_working" = "yes"; then
-        ac_cv_pthreads_cflags=$flag
-        break
-      fi
-    done
-  fi
-fi
-])
-
-AC_CACHE_CHECK(for pthreads_lib, ac_cv_pthreads_lib,[
-ac_cv_pthreads_lib=
-if test "$pthreads_working" != "yes"; then
-  for lib in pthread pthreads c_r; do
-    ac_save=$LIBS
-    LIBS="$LIBS -l$lib"
-    PTHREADS_CHECK_COMPILE
-    LIBS=$ac_save
-    if test "$pthreads_working" = "yes"; then
-      ac_cv_pthreads_lib=$lib
-      break
-    fi
-  done
-fi
-])
+pthreads_working="yes"
+ac_cv_pthreads_cflags="-pthread"
 
 if test "$pthreads_working" = "yes"; then
   threads_result="POSIX-Threads found"
