From: =?utf-8?b?T25kxZllaiBTdXLDvQ==?= <ondrej@sury.org>
Date: Sun, 17 Aug 2025 08:49:33 +0200
Subject: Fix AC_PROG_SED error and other autoconf warnings

---
 acinclude.m4      | 6 +++---
 aclocal.m4        | 6 +++---
 scripts/phpize.m4 | 4 ++--
 3 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/acinclude.m4 b/acinclude.m4
index fffa0a9..6b5544b 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -303,7 +303,7 @@ AC_MSG_CHECKING([if compiler supports -R])
 AC_CACHE_VAL(php_cv_cc_dashr,[
   SAVE_LIBS=$LIBS
   LIBS="-R /usr/$PHP_LIBDIR $LIBS"
-  AC_TRY_LINK([], [], php_cv_cc_dashr=yes, php_cv_cc_dashr=no)
+  AC_LINK_IFELSE([AC_LANG_PROGRAM([], [])],[php_cv_cc_dashr=yes],[php_cv_cc_dashr=no])
   LIBS=$SAVE_LIBS])
 AC_MSG_RESULT([$php_cv_cc_dashr])
 if test $php_cv_cc_dashr = "yes"; then
@@ -313,7 +313,7 @@ else
   AC_CACHE_VAL(php_cv_cc_rpath,[
     SAVE_LIBS=$LIBS
     LIBS="-Wl,-rpath,/usr/$PHP_LIBDIR $LIBS"
-    AC_TRY_LINK([], [], php_cv_cc_rpath=yes, php_cv_cc_rpath=no)
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([], [])],[php_cv_cc_rpath=yes],[php_cv_cc_rpath=no])
     LIBS=$SAVE_LIBS])
   AC_MSG_RESULT([$php_cv_cc_rpath])
   if test $php_cv_cc_rpath = "yes"; then
@@ -2678,7 +2678,7 @@ dnl Generates the config.nice file
 dnl
 AC_DEFUN([PHP_CONFIG_NICE],[
   AC_REQUIRE([AC_PROG_EGREP])
-  AC_REQUIRE([LT_AC_PROG_SED])
+  AC_REQUIRE([AC_PROG_SED])
   PHP_SUBST_OLD(EGREP)
   PHP_SUBST_OLD(SED)
   test -f $1 && mv $1 $1.old
diff --git a/aclocal.m4 b/aclocal.m4
index d07dcb0..49020bf 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -303,7 +303,7 @@ AC_MSG_CHECKING([if compiler supports -R])
 AC_CACHE_VAL(php_cv_cc_dashr,[
   SAVE_LIBS=$LIBS
   LIBS="-R /usr/$PHP_LIBDIR $LIBS"
-  AC_TRY_LINK([], [], php_cv_cc_dashr=yes, php_cv_cc_dashr=no)
+  AC_LINK_IFELSE([AC_LANG_PROGRAM([], [])],[php_cv_cc_dashr=yes],[php_cv_cc_dashr=no])
   LIBS=$SAVE_LIBS])
 AC_MSG_RESULT([$php_cv_cc_dashr])
 if test $php_cv_cc_dashr = "yes"; then
@@ -313,7 +313,7 @@ else
   AC_CACHE_VAL(php_cv_cc_rpath,[
     SAVE_LIBS=$LIBS
     LIBS="-Wl,-rpath,/usr/$PHP_LIBDIR $LIBS"
-    AC_TRY_LINK([], [], php_cv_cc_rpath=yes, php_cv_cc_rpath=no)
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([], [])],[php_cv_cc_rpath=yes],[php_cv_cc_rpath=no])
     LIBS=$SAVE_LIBS])
   AC_MSG_RESULT([$php_cv_cc_rpath])
   if test $php_cv_cc_rpath = "yes"; then
@@ -3673,7 +3673,7 @@ s390*-*linux*|sparc*-*linux*)
   AC_CACHE_CHECK([whether the C compiler needs -belf], lt_cv_cc_needs_belf,
     [AC_LANG_SAVE
      AC_LANG_C
-     AC_TRY_LINK([],[],[lt_cv_cc_needs_belf=yes],[lt_cv_cc_needs_belf=no])
+     AC_LINK_IFELSE([AC_LANG_PROGRAM([], [])],[lt_cv_cc_needs_belf=yes],[lt_cv_cc_needs_belf=no])
      AC_LANG_RESTORE])
   if test x"$lt_cv_cc_needs_belf" != x"yes"; then
     # this is probably gcc 2.8.0, egcs 1.0 or newer; no need for -belf
diff --git a/scripts/phpize.m4 b/scripts/phpize.m4
index d745ca7..e3f9b5f 100644
--- a/scripts/phpize.m4
+++ b/scripts/phpize.m4
@@ -146,7 +146,7 @@ AC_PROVIDE_IFELSE([PHP_REQUIRE_CXX], [], [
   undefine([AC_PROG_CXXCPP])
   AC_DEFUN([AC_PROG_CXXCPP], [php_prog_cxxcpp=disabled])
 ])
-AC_PROG_LIBTOOL
+LT_INIT
 
 all_targets='$(PHP_MODULES) $(PHP_ZEND_EX)'
 install_targets="install-modules install-headers"
@@ -197,6 +197,6 @@ PHP_GEN_GLOBAL_MAKEFILE
 test -d modules || $php_shtool mkdir modules
 touch .deps
 
-AC_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 
 AC_OUTPUT()
