From: =?utf-8?b?T25kxZllaiBTdXLDvQ==?= <ondrej@sury.org>
Date: Sun, 18 Oct 2020 13:31:37 +0200
Subject: Include all libtool files from phpize.m4

---
 scripts/phpize.m4 | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/scripts/phpize.m4 b/scripts/phpize.m4
index be7065b..6fb4f33 100644
--- a/scripts/phpize.m4
+++ b/scripts/phpize.m4
@@ -5,6 +5,10 @@ dnl comments starting with # and empty newlines from the included files.
 m4_include([build/ax_check_compile_flag.m4])
 m4_include([build/ax_gcc_func_attribute.m4])
 m4_include([build/libtool.m4])
+m4_include([build/ltoptions.m4])
+m4_include([build/ltsugar.m4])
+m4_include([build/ltversion.m4])
+m4_include([build/lt~obsolete.m4])
 m4_include([build/php_cxx_compile_stdcxx.m4])
 m4_include([build/php.m4])
 m4_include([build/pkg.m4])
