From: Debian PHP Maintainers <pkg-php-maint@lists.alioth.debian.org>
Date: Sat, 2 May 2015 10:26:57 +0200
Subject: hack-phpdbg-to-explicitly-link-with-libedit

---
 sapi/phpdbg/config.m4 | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sapi/phpdbg/config.m4 b/sapi/phpdbg/config.m4
index d4bceed..ad542f5 100644
--- a/sapi/phpdbg/config.m4
+++ b/sapi/phpdbg/config.m4
@@ -42,7 +42,9 @@ if test "$BUILD_PHPDBG" = "" && test "$PHP_PHPDBG" != "no"; then
   if test "$PHP_PHPDBG_READLINE" = "yes"; then
     if test "$PHP_READLINE" != "no" -o  "$PHP_LIBEDIT" != "no"; then
   	  AC_DEFINE(HAVE_PHPDBG_READLINE, 1, [ ])
-  	  PHPDBG_EXTRA_LIBS="$PHP_READLINE_LIBS"
+#  	  PHPDBG_EXTRA_LIBS="$PHP_READLINE_LIBS"
+# Temporary hack until this gets fixed
+  	  PHPDBG_EXTRA_LIBS="-ledit -ltermcap"
   	  AC_MSG_RESULT([ok])
   	else
   	  AC_MSG_RESULT([readline is not available])
