From: Remi Collet <remi@remirepo.net>
Date: Tue, 6 Jun 2023 18:05:22 +0200
Subject: Fix GH-11382 add missing hash header for bin2hex

---
 ext/soap/php_http.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ext/soap/php_http.c b/ext/soap/php_http.c
index e3a9afd..912b8e3 100644
--- a/ext/soap/php_http.c
+++ b/ext/soap/php_http.c
@@ -22,6 +22,7 @@
 #include "ext/standard/base64.h"
 #include "ext/standard/md5.h"
 #include "ext/standard/php_random.h"
+#include "ext/hash/php_hash.h"
 
 static char *get_http_header_value_nodup(char *headers, char *type, size_t *len);
 static char *get_http_header_value(char *headers, char *type);
