From: Debian PHP Maintainers <pkg-php-maint@lists.alioth.debian.org>
Date: Sat, 2 May 2015 10:26:54 +0200
Subject: hurd-noptrace

---
 sapi/fpm/config.m4 | 5 +++++
 1 file changed, 5 insertions(+)

--- a/sapi/fpm/config.m4
+++ b/sapi/fpm/config.m4
@@ -141,6 +141,11 @@
         pid_t child;
         int status;
 
+        /* broken ptrace on Hurd, avoid hanging */
+        #ifdef __GNU__
+        return 10;
+        #endif
+
         if ( (child = fork()) ) { /* parent */
           int ret = 0;
 
