From: Remi Collet <remi@remirepo.net>
Date: Wed, 10 Apr 2024 16:18:48 +0200
Subject: NEWS

---
 NEWS | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/NEWS b/NEWS
index 8b4801d..1300609 100644
--- a/NEWS
+++ b/NEWS
@@ -8,6 +8,8 @@ Backported from 8.1.28
     partial CVE-2022-31629 fix). (CVE-2024-2756) (nielsdos)
   . Fixed bug GHSA-h746-cjrr-wfmr (password_verify can erroneously return true,
     opening ATO risk). (CVE-2024-3096) (Jakub Zelenka)
+  . Fixed bug GHSA-pc52-254m-w9w7 (Command injection via array-ish $command
+    parameter of proc_open). (CVE-2024-1874) (Jakub Zelenka)
 
 03 Aug 2023, PHP 8.0.30
 
