From: =?utf-8?b?T25kxZllaiBTdXLDvQ==?= <ondrej@sury.org>
Date: Tue, 11 Mar 2025 17:26:16 +0100
Subject: Add #ifdef guards to main/php_config.h.in

---
 configure.in | 29 +++++++++++++++++------------
 1 file changed, 17 insertions(+), 12 deletions(-)

diff --git a/configure.in b/configure.in
index 51bb70e..ba91a73 100644
--- a/configure.in
+++ b/configure.in
@@ -18,6 +18,9 @@ PHP_CANONICAL_HOST_TARGET
 
 AC_CONFIG_HEADER(main/php_config.h)
 AH_TOP([
+#ifndef PHP_CONFIG_H
+#define PHP_CONFIG_H
+
 #if defined(__GNUC__) && __GNUC__ >= 4
 # define ZEND_API __attribute__ ((visibility("default")))
 # define ZEND_DLEXPORT __attribute__ ((visibility("default")))
@@ -115,6 +118,8 @@ int zend_sprintf(char *buffer, const char *format, ...);
 #endif
 
 #undef PTHREADS
+
+#endif /* PHP_CONFIG_H */
 ])
 
 PHP_MAJOR_VERSION=7
@@ -173,7 +178,7 @@ if test -n "$with_apxs2filter" && test -n "$with_apxs2"; then
   AC_MSG_ERROR([--with-apxs2filter and --with-apxs2 cannot be used together])
 fi
 
-  
+
 dnl Settings we want to make before the checks.
 dnl -------------------------------------------------------------------------
 
@@ -709,7 +714,7 @@ dnl Check for getaddrinfo, should be a better way, but...
 dnl Also check for working getaddrinfo
 AC_CACHE_CHECK([for getaddrinfo], ac_cv_func_getaddrinfo,
 [AC_TRY_LINK([#include <netdb.h>],
-                [struct addrinfo *g,h;g=&h;getaddrinfo("","",g,&g);], 
+                [struct addrinfo *g,h;g=&h;getaddrinfo("","",g,&g);],
   AC_TRY_RUN([
 #include <netdb.h>
 #include <sys/types.h>
@@ -731,7 +736,7 @@ int main(void) {
   }
 
   pai = ai;
-  
+
   while (pai) {
     if (pai->ai_family != AF_INET) {
       /* 127.0.0.1/NUMERICHOST should only resolve ONE way */
@@ -788,7 +793,7 @@ if test "$PHP_GCOV" = "yes"; then
   if test "$GCC" != "yes"; then
     AC_MSG_ERROR([GCC is required for --enable-gcov])
   fi
-  
+
   dnl Check if ccache is being used
   case `$php_shtool path $CC` in
     *ccache*[)] gcc_ccache=yes;;
@@ -798,11 +803,11 @@ if test "$PHP_GCOV" = "yes"; then
   if test "$gcc_ccache" = "yes" && (test -z "$CCACHE_DISABLE" || test "$CCACHE_DISABLE" != "1"); then
     AC_MSG_ERROR([ccache must be disabled when --enable-gcov option is used. You can disable ccache by setting environment variable CCACHE_DISABLE=1.])
   fi
-  
+
   dnl min: 1.5 (i.e. 105, major * 100 + minor for easier comparison)
   ltp_version_min="105"
   dnl non-working versions, e.g. "1.8 1.18";
-  dnl remove "none" when introducing the first incompatible LTP version an 
+  dnl remove "none" when introducing the first incompatible LTP version an
   dnl separate any following additions by spaces
   ltp_version_exclude="1.8"
 
@@ -831,7 +836,7 @@ if test "$PHP_GCOV" = "yes"; then
       fi
     ])
   else
-    ltp_msg="To enable code coverage reporting you must have LTP installed"      
+    ltp_msg="To enable code coverage reporting you must have LTP installed"
     AC_MSG_ERROR([$ltp_msg])
   fi
 
@@ -921,7 +926,7 @@ fi
 AC_MSG_CHECKING([where to scan for configuration files])
 PHP_ARG_WITH(config-file-scan-dir,,
 [  --with-config-file-scan-dir=PATH
-                          Set the path where to scan for configuration files], DEFAULT, no) 
+                          Set the path where to scan for configuration files], DEFAULT, no)
 if test "$PHP_CONFIG_FILE_SCAN_DIR" = "DEFAULT"; then
   PHP_CONFIG_FILE_SCAN_DIR=
 fi
@@ -1019,7 +1024,7 @@ dnl -------------------------------------------------------------------------
 PHP_HELP_SEPARATOR([Extensions:
 
   --with-EXTENSION=[shared[,PATH]]
-  
+
     NOTE: Not all extensions can be build as 'shared'.
 
     Example: --with-foobar=shared,/usr/local/foobar/
@@ -1106,13 +1111,13 @@ if test "$PHP_PEAR" != "no"; then
   dnl PEAR dependancies
   dnl
   if test "$PHP_XML" = "no"; then
-    pear_error_msg="$pear_error_msg 
+    pear_error_msg="$pear_error_msg
                     PEAR requires XML to be enabled.     Add --enable-xml to the configure line. (or --without-pear)"
   fi
 
 dnl
 dnl  if test "$PHP_XMLRPC" = "no"; then
-dnl    pear_error_msg="$pear_error_msg 
+dnl    pear_error_msg="$pear_error_msg
 dnl                    PEAR requires XML-RPC to be enabled. Add --with-xmlrpc to the configure line. (or --without-pear)"
 dnl  fi
 dnl
@@ -1230,7 +1235,7 @@ if test -z "$EXTENSION_DIR"; then
     if test "$enable_maintainer_zts" = "yes"; then
       extbasedir=$extbasedir-zts
     fi
-    
+
     if test "$PHP_DEBUG" = "1"; then
       extbasedir=$extbasedir-debug
     fi
