From: Thijs Kinkhorst <thijs@debian.org>
Date: Wed, 15 Jun 2016 09:18:03 +0200
Subject: Add patch to install php7 module directly to APXS_LIBEXEC

---
 sapi/apache2handler/config.m4 | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/sapi/apache2handler/config.m4 b/sapi/apache2handler/config.m4
index e335721..e754933 100644
--- a/sapi/apache2handler/config.m4
+++ b/sapi/apache2handler/config.m4
@@ -69,18 +69,9 @@ if test "$PHP_APXS2" != "no"; then
     [AC_MSG_ERROR([Please note that Apache version >= 2.4 is required])])
 
   APXS_LIBEXECDIR='$(INSTALL_ROOT)'$($APXS -q LIBEXECDIR)
-  if test -z $($APXS -q SYSCONFDIR); then
-    INSTALL_IT="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
-                 $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
-                       -i -n php"
-  else
-    APXS_SYSCONFDIR='$(INSTALL_ROOT)'$($APXS -q SYSCONFDIR)
-    INSTALL_IT="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
-                \$(mkinstalldirs) '$APXS_SYSCONFDIR' && \
-                 $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
-                       -S SYSCONFDIR='$APXS_SYSCONFDIR' \
-                       -i -a -n php"
-  fi
+  INSTALL_IT="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
+               $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
+                     -i -n php"
 
   LIBPHP_CFLAGS="-shared"
   PHP_SUBST([LIBPHP_CFLAGS])
