From 8c4a093917e08d3e76dda232431c121837dd354b Mon Sep 17 00:00:00 2001
From: Shivam Mathur <shivam_jpr@hotmail.com>
Date: Sun, 15 Mar 2026 08:40:10 +0530
Subject: [PATCH] libtool2.2

---
 scripts/phpize.in | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/scripts/phpize.in b/scripts/phpize.in
index 7ebd1c8e06d..4f1939eec1d 100644
--- a/scripts/phpize.in
+++ b/scripts/phpize.in
@@ -9,9 +9,11 @@ includedir="$prefix/include/php/@DEBIAN_PHP_API@"
 builddir="`pwd`"
 SED="@SED@"
 
-FILES_BUILD="php.m4 shtool libtool.m4 ax_check_compile_flag.m4 ax_gcc_func_attribute.m4 php_cxx_compile_stdcxx.m4 pkg.m4 \
-    config.guess config.sub ltmain.sh lt~obsolete.m4 ltoptions.m4 ltsugar.m4 ltversion.m4 Makefile.global gen_stub.php"
+aclocaldir="$prefix/share/aclocal"
+FILES_BUILD="php.m4 shtool ax_check_compile_flag.m4 ax_gcc_func_attribute.m4 php_cxx_compile_stdcxx.m4 \
+     config.guess config.sub ltmain.sh Makefile.global gen_stub.php"
 FILES="run-tests*.php"
+LIBTOOL_FILES="pkg.m4 libtool.m4 ltoptions.m4 ltsugar.m4 ltversion.m4 lt~obsolete.m4"
 CLEAN_FILES="$FILES *.o *.lo *.la .libs/ build/ modules/ \
   config.nice configure configure.ac \
   config.h config.h.in conftest* libtool config.cache autom4te.cache/ \
@@ -149,6 +151,7 @@ phpize_copy_files()
   test -d build || mkdir build
 
   (cd "$phpdir" && cp $FILES_BUILD "$builddir"/build)
+  (cd "$aclocaldir" && cp $LIBTOOL_FILES "$builddir"/build)
   (cd "$phpdir" && cp $FILES "$builddir")
 }
 
-- 
2.50.1 (Apple Git-155)

